src/Engine/include/Engine/UI/UiWindow.hpp

```diff
@@ -25,30 +31,146 @@ namespace Engine {
+
+        /**
+         * @brief Close the window
+         * 
+         */
         auto close()    -> void { m_isOpen = false; }
+
+        /**
+         * @brief Render the window and everything it contains
+         * 
+         */
         virtual auto render()   -> void;
-        
+
+        /**
+         * @brief Set the Window Flags object
+         * 
+         * @param flags The value of the flags, different value will trigger different options
+         */
         auto setWindowFlags(ImGuiWindowFlags flags) noexcept -> void { m_window_flags = flags; }
+
+        /**
+         * @brief Set the Position object
+         * 
+         * @param pos The new postion of the object
+         */
         auto setPosition(const ImVec2 &pos) noexcept    -> void { m_position = pos; }
+
+        /**
+         * @brief Set the Size object
+         * 
+         * @param size The new size of the object
+         */
         auto setSize(const ImVec2 &size)    noexcept    -> void { m_size = size; }
+
+        /**
+         * @brief Set the Name object
+         * 
+         * @param name The new name of the object
+         */
         auto setName(const std::string_view &name) noexcept  -> void { m_name = name; }
 
+        /**
+         * @brief Get the Flags object
+         * 
+         * @return ImGuiWindowFlags& 
+         */
         auto getFlags()     -> ImGuiWindowFlags &{ return m_window_flags; }
+
+        /**
+         * @brief Get the Position object
+         * 
+         * @return const ImVec2& 
+         */
         auto getPosition()  -> const ImVec2 & { return m_position; }
+        /**
+         * @brief Get the Size object
+         * 
+         * @return const ImVec2& 
+         */
         auto getSize()      -> const ImVec2 & { return m_size; }
+        /**
+         * @brief Get the Name object
+         * 
+         * @return const std::string_view& 
+         */
         auto getName()      -> const std::string_view & { return m_name; }
 
-        auto addWidget(std::shared_ptr<AWidget> wid) -> void { m_WidgetList.emplace_back(wid); }
+        /**
+         * @brief Add a new widget to the window
+         * 
+         * @param widget The new widget to add 
+         */
+        auto addWidget(std::shared_ptr<AWidget> widget) -> void { m_WidgetList.emplace_back(widget); }
+        
+        /**
+         * @brief Delete a widget store in the window
+         * 
+         * @param name The name of the widget to delete
+         * @return true Return true if a widget was delete
+         * @return false Return false if no widget was found
+         */
         auto deleteWidget(const std::string_view &name)  -> bool;
+        
+        /**
+         * @brief Get the Widget object
+         * 
+         * @param name The name of the widget to get
+         * @return std::optional<std::shared_ptr<AWidget>> Return a widget smart pointer if found
+         * @return nullopt Return nullopt if the widget was not found
```

Comment:
Something I didn't catch in the first PR, `std::optional` is not needed here because you could return `nullptr` for the exact same result.